runtime/src/sorted_storages.rs

```diff
@@ -0,0 +1,117 @@
+use crate::accounts_db::SnapshotStorage;
+use log::*;
+use solana_measure::measure::Measure;
+use solana_sdk::clock::Slot;
+use std::ops::Range;
+
+pub struct SortedStorages<'a> {
+    range: Range<Slot>,
+    storages: Vec<Option<&'a SnapshotStorage>>,
+    count: usize,
+}
+
+impl<'a> SortedStorages<'a> {
+    pub fn get(&self, slot: Slot) -> Option<&SnapshotStorage> {
+        if !self.range.contains(&slot) {
+            None
+        } else {
+            let index = (slot - self.range.start) as usize;
+            self.storages[index]
+        }
+    }
+
+    pub fn range_width(&self) -> u64 {
```

Comment:
nit: Should this return value be `Slot` instead? I don't have strong feeling about this, just noticed it was different.

Reply:
done